velox/vector/arrow/tests/ArrowBridgeSchemaTest.cpp

```diff
@@ -433,5 +455,62 @@ TEST_F(ArrowBridgeSchemaTest, validateInArrow) {
   }
 }
 
+TEST_F(ArrowBridgeSchemaImportTest, dictionaryTypeTest) {
+  // Primitive types
+  EXPECT_EQ(DOUBLE(), testSchemaDictionaryImport("i", makeArrowSchema("g")));
+  EXPECT_EQ(BOOLEAN(), testSchemaDictionaryImport("i", makeArrowSchema("b")));
+  EXPECT_EQ(TINYINT(), testSchemaDictionaryImport("i", makeArrowSchema("c")));
+  EXPECT_EQ(INTEGER(), testSchemaDictionaryImport("i", makeArrowSchema("i")));
+  EXPECT_EQ(SMALLINT(), testSchemaDictionaryImport("i", makeArrowSchema("s")));
+  EXPECT_EQ(BIGINT(), testSchemaDictionaryImport("i", makeArrowSchema("l")));
+  EXPECT_EQ(REAL(), testSchemaDictionaryImport("i", makeArrowSchema("f")));
+  EXPECT_EQ(VARCHAR(), testSchemaDictionaryImport("i", makeArrowSchema("u")));
+
+  std::vector<ArrowSchema> schemas;
+  std::vector<ArrowSchema*> schemaPtrs;
+
+  // Arrays
+  EXPECT_EQ(
+      *ARRAY(BIGINT()),
+      *testSchemaDictionaryImport(
+          "i", makeComplexArrowSchema(schemas, schemaPtrs, "+l", {"l"})));
+  EXPECT_EQ(
+      *ARRAY(TIMESTAMP()),
+      *testSchemaDictionaryImport(
+          "i", makeComplexArrowSchema(schemas, schemaPtrs, "+l", {"ttn"})));
+  EXPECT_EQ(
+      *ARRAY(DATE()),
+      *testSchemaDictionaryImport(
+          "i", makeComplexArrowSchema(schemas, schemaPtrs, "+l", {"tdD"})));
+  EXPECT_EQ(
+      *ARRAY(VARCHAR()),
+      *testSchemaDictionaryImport(
+          "i", makeComplexArrowSchema(schemas, schemaPtrs, "+l", {"U"})));
+
+  // Maps
+  EXPECT_EQ(
+      *MAP(VARCHAR(), BOOLEAN()),
+      *testSchemaDictionaryImport(
+          "i", makeComplexArrowSchema(schemas, schemaPtrs, "+m", {"U", "b"})));
+  EXPECT_EQ(
+      *MAP(SMALLINT(), REAL()),
+      *testSchemaDictionaryImport(
+          "i", makeComplexArrowSchema(schemas, schemaPtrs, "+m", {"s", "f"})));
+
+  // Rows
+  EXPECT_EQ(
+      *ROW({SMALLINT(), REAL()}),
+      *testSchemaDictionaryImport(
+          "i", makeComplexArrowSchema(schemas, schemaPtrs, "+s", {"s", "f"})));
+
+  // Named Row
+  EXPECT_EQ(
+      *ROW({"col1", "col2"}, {SMALLINT(), REAL()}),
+      *testSchemaDictionaryImport(
+          "i",
+          makeComplexArrowSchema(
+              schemas, schemaPtrs, "+s", {"s", "f"}, {"col1", "col2"})));
```

Comment:
can we have nested dictionary types (dict(dict(integer)))? If so, do we have tests for them?